### ISSUE TITLE
Getting error in isomorphic app

### ISSUE THREAD

**Issue by Anahkiasen** [zalmoxisus/redux-devtools-extension#125]
Since today I'm getting the following error when trying to load my app:

![screen shot 2016-05-30 at 11 59 48](https://cloud.githubusercontent.com/assets/1321596/15646477/1caa14cc-265e-11e6-92b1-0b3dbafb6f12.png)

This is my middleware:

devtools.js
```js
export default (typeof window === 'object' && typeof window.devToolsExtension !== 'undefined')
    ? window.devToolsExtension({actionsBlacklist: ['redux-form/FOCUS', 'redux-form/BLUR']})
    : state => state;
```

And this is my store:

Store/index.js
```js
import {createStore} from 'redux';
import reducers from '../Reducers';
import devtools from './Middlewares/Devtools';

export default function (initialState) {
    return createStore(reducers, initialState, devtools);
}
```

I checked and `reducers` is a function, and if I do `reducers({}, {type: 'FOOBAR'})` I get my state back, correct shape and all. Am I doing something wrong here? Note that if I remove the devtools the app runs fine

I'm running v2 of the extension, which got shipped May 28th, didn't have the issue before.

**Comment by Anahkiasen** [comment]
On a related note I can't actually find 2.0 anywhere, repos is still at 1.4, where is it?

**Comment by Arilas** [comment]
Receive the same error, after look at the devToolsExtenstion.toString() found this:
```js
function(reducer, initialState, config) { /* eslint-disable no-param-reassign */
    if ((typeof reducer === 'undefined' ? 'undefined' : _typeof(reducer)) === 'object') config = reducer;
    else if ((typeof config === 'undefined' ? 'undefined' : _typeof(config)) !== 'object') config = {}; /* eslint-enable no-param-reassign */
    if (!window.devToolsOptions) window.devToolsOptions = {};
    var store = void 0;
    var liftedStore = void 0;
    var shouldSerialize = false;
    var lastAction = void 0;
    var errorOccurred = false;
    var isMonitored = false;
    var isExcess = void 0;
    var instanceId = (0, _contentScriptMsg.generateId)(config.instanceId);
    var localFilter = (0, _filters.getLocalFilter)(config);
    var _config = config;
    var statesFilter = _config.statesFilter;
    var actionsFilter = _config.actionsFilter;

    function relay(type, state, action, nextActionId) {
        var message = {
            type: type,
            payload: (0, _filters.filterState)(state, type, localFilter, statesFilter, actionsFilter, nextActionId),
            source: '@devtools-page',
            id: instanceId
        };
        if (type === 'ACTION') {
            message.action = !actionsFilter ? action : actionsFilter(action.action, nextActionId - 1);
            message.isExcess = isExcess;
            message.nextActionId = nextActionId;
        } else if (action) {
            message.action = action;
        } else {
            message.name = config.name || document.title;
        }
        if (shouldSerialize || window.devToolsOptions.serialize) {
            (0, _contentScriptMsg.toContentScript)(message, true);
        } else {
            try {
                (0, _contentScriptMsg.toContentScript)(message);
            } catch (err) {
                (0, _contentScriptMsg.toContentScript)(message, true);
                shouldSerialize = true;
            }
        }
    }

    function onMessage(message) {
        switch (message.type) {
            case 'DISPATCH':
                liftedStore.dispatch(message.payload);
                return;
            case 'ACTION':
                store.dispatch(message.payload);
                return;
            case 'IMPORT':
                var nextLiftedState = (0, _importState2.default)(message.state, config);
                if (!nextLiftedState) return;
                liftedStore.dispatch({
                    type: 'IMPORT_STATE',
                    nextLiftedState: nextLiftedState
                });
                relay('STATE', liftedStore.getState());
                return;
            case 'UPDATE':
                relay('STATE', liftedStore.getState());
                return;
            case 'START':
                isMonitored = true;
                relay('STATE', liftedStore.getState());
                return;
            case 'STOP':
                isMonitored = false;
        }
    }

    function init() {
        (0, _contentScriptMsg.setListener)(onMessage, instanceId);
        (0, _notifyErrors2.default)(function() {
            errorOccurred = true;
            var state = liftedStore.getState();
            if (state.computedStates[state.currentStateIndex].error) {
                relay('STATE', state);
                return false;
            }
            return true;
        });
        relay('INIT_INSTANCE');
    }

    function monitorReducer() {
        var state = arguments.length <= 0 || arguments[0] === undefined ? {} : arguments[0];
        var action = arguments[1];
        if (!isMonitored) return state;
        lastAction = action.type;
        if (lastAction === '@@redux/INIT' && liftedStore) { // Send new lifted state on hot-reloading
            setTimeout(function() {
                relay('STATE', liftedStore.getState());
            }, 0);
        }
        return state;
    }
    var monitorActions = ['TOGGLE_ACTION', 'SWEEP', 'SET_ACTIONS_ACTIVE', 'IMPORT_STATE'];

    function handleChange(state, liftedState) {
        if (!isMonitored) return;
        var nextActionId = liftedState.nextActionId;
        var liftedAction = liftedState.actionsById[nextActionId - 1];
        var action = liftedAction.action;
        if (action.type === '@@INIT') {
            relay('INIT', state, {
                timestamp: Date.now()
            });
        } else if (!errorOccurred && monitorActions.indexOf(lastAction) === -1) {
            if (lastAction === 'JUMP_TO_STATE' || (0, _filters.isFiltered)(action, localFilter)) return;
            var maxAge = window.devToolsOptions.maxAge;
            relay('ACTION', state, liftedAction, nextActionId);
            if (!isExcess && maxAge) isExcess = liftedState.stagedActionIds.length >= maxAge;
        } else {
            if (errorOccurred && !liftedState.computedStates[liftedState.currentStateIndex].error) {
                errorOccurred = false;
            }
            relay('STATE', liftedState);
        }
    }
    var enhance = function enhance() {
        return function(next) {
            return function(reducer_, initialState_, enhancer_) {
                if (!(0, _syncOptions.isAllowed)(window.devToolsOptions)) return next(reducer_, initialState_, enhancer_);
                store = (0, _configureStore2.default)(next, monitorReducer, config)(reducer_, initialState_, enhancer_);
                liftedStore = store.liftedStore;
                init();
                store.subscribe(function() {
                    if (liftedStore !== store.liftedStore) liftedStore = store.liftedStore;
                    handleChange(store.getState(), liftedStore.getState());
                });
                return store;
            };
        };
    };
    if (!reducer) return enhance();
    return (0, _createStore2.default)(reducer, initialState, enhance);
}
```
You can see that extension now contains redux(_createStore2.default), also you can see always false condition:
```js
if (!reducer) return enhance();
```
So for now extension doesn't work with first parameter as options, workaround:
```js
window.devToolsExtension(undefined, undefined, options)
```

**Comment by Arilas** [comment]
Seems that author want to use his extension like replacement of real redux

**Comment by Anahkiasen** [comment]
Can confirm, if I don't pass any arguments to `devToolsExtension` it works fine.

**Comment by zalmoxisus** [comment]
Sorry guys, we should definitely add tests!

I just published `2.0.0.1` with the fix to Chrome Store. It would be available in about an hour.

@Arilas, yes you nailed it! I'll document later this approach, though it's shown in [the example](https://github.com/zalmoxisus/redux-devtools-extension/blob/master/examples/react-counter/components/Counter.js).
